analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -582,6 +663,256 @@ void AsOperatorWithNonNullLiteral(int? i)
     }
 }
 
+class OutAndRefParams
+{
+    int? nullableField;
+    readonly int? nullableReadonlyField = null;
+
+    OutAndRefParams()
+    {
+        nullableReadonlyField = null;
+        ModifyOutParamInCtor(out nullableReadonlyField);
+        _ = nullableReadonlyField.Value;  // Compliant
+
+        static void ModifyOutParamInCtor(out int? i) => i = null;
+    }
+
+    void OutParams(int? iParam)
+    {
+        iParam = null;
+        ModifyOutParam(out iParam);
+        _ = iParam.Value;                                // Compliant, unknown after method call
+
+        int? iLocal;
+        ModifyOutParam(out iLocal);
+        _ = iLocal.Value;                                // Compliant
+
+        iLocal = null;
+        ModifyOutParam(out iLocal);
+        _ = iLocal.Value;                                // Compliant
+
+        iLocal = null;
+        ModifyOutParamAndRead(out iLocal, iLocal.Value); // FN
+
+        iLocal = null;
+        ReadAndModifyOutParam(iLocal.Value, out iLocal); // Noncompliant
+
+        nullableField = null;
+        ModifyOutParam(out nullableField);
+        _ = nullableField.Value;                         // Compliant
+
+        ModifyOutParam(out var newLocal);
+        _ = newLocal.Value;                             // Unknown
+        newLocal = null;
+        _ = newLocal.Value;                             // Noncompliant
+
+        static void ModifyOutParam(out int? i) => i = null;
+        static void ModifyOutParamAndRead(out int? i1, int? i2) => i1 = i2;
+        static void ReadAndModifyOutParam(int? i1, out int? i2) => i2 = i1;
+    }
+
+    void RefParams(int? iParam)
+    {
+        iParam = null;
+        ModifyRefParam(ref iParam);
+        _ = iParam.Value;                                // Compliant, unknown after method call
+
+        int? iLocal = null;
+        ModifyRefParam(ref iLocal);
+        _ = iLocal.Value;                                // Compliant
+
+        iLocal = null;
+        ModifyRefParamAndRead(ref iLocal, iLocal.Value); // FN
+
+        iLocal = null;
+        ReadAndModifyRefParam(iLocal.Value, ref iLocal); // Noncompliant
+
+        nullableField = null;
+        ModifyRefParam(ref nullableField);
+        _ = nullableField.Value;                         // Compliant
+
+        static void ModifyRefParam(ref int? i) => i = null;
+        static void ModifyRefParamAndRead(ref int? i1, int? i2) => i1 = i2;
+        static void ReadAndModifyRefParam(int? i1, ref int? i2) => i2 = i1;
+    }
```

Comment:
Can we get examples with [`in`-modifier](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-parameter-modifier)? One or two simple cases should be enough.